File: api/src/main/java/net/blueshell/api/dto/GuestDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = false)
@Schema(name = "Guest")
public class GuestDTO extends PersonalInfoDTO {
    private Long id;
    @NotNull
    private String name;
    private String accessToken;
}

File: api/src/main/kotlin/net/blueshell/api/dto/GuestDTO.kt
package net.blueshell.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull

@Schema(name = "Guest")
data class GuestDTO(
    @field:NotNull
    var name: String? = null,
) : PersonalInfoDTO() {
    lateinit var accessToken: String
}
